.github/workflows/consumer_contract_tests.yml

```diff
@@ -26,8 +26,42 @@ env:
   CAN_I_DEPLOY_RUN_NAME: 'can-i-deploy-${{ github.event.repository.name }}-${{ github.run_id }}-${{ github.run_attempt }}'
 
 jobs:
+  bump-check:
+    runs-on: ubuntu-latest
+    outputs:
+      is-bump: ${{ steps.skiptest.outputs.is-bump }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Skip version bump merges
+        id: skiptest
+        uses: ./.github/actions/bump-skip
+        with:
+          event-name: ${{ github.event_name }}
+
+# We regulate the tag job to meet the following requirements according to the trigger event type:
+# 1. pull_request event (due to opening or updating of PR branch):
```

Comment:
Reading this and scratching my head in confusion...

Shouldn't tags only get created for real when a merge to main happens?

I'm getting a bit hung up `dry-run` here being used for what seems like the exact opposite of my understanding of what a dry-run is for.

IMO, anything that's on a branch and is _not_ main is considered dry-run territory, not the opposite, as used here.

Reply:
Just saw your explanation to this effect in Slack, which I think illuminates things a bit more.  I wonder if this text would help if included in the comment too?

"The key part about this PR is that the actual tag should only be dispatched during "Tag, publish, deploy" workflow, but because contract tests need to know the new tag in advance before actual deployment takes place, it relies on the bumper's dry run feature to get the next tag to publish contracts and results from verifying consumer pacts, without actually dispatching the tag. Otherwise, we need to orchestrate multiple workflows and that means a larger restructuring will need to take place. This is because Sherlock use semver instead of git hash for deployment."

Reply:
The bumper doc [](https://github.com/DataBiosphere/github-actions/tree/master/actions/bumper) has this definition

```
DRY_RUN (optional) - Determine the next version without tagging the branch. The workflow can use the outputs new_tag and tag in subsequent steps. Possible values are true and false (default).
```

Reply:
More comments have been added to `regulated-tag-job` section.